src/Razor/src/Microsoft.NET.Sdk.Razor/ValidateStaticWebAssetsUniquePaths.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class ValidateStaticWebAssetsUniquePaths : Task
+    {
+        private const string BasePath = "BasePath";
+        private const string RelativePath = "RelativePath";
+        private const string TargetPath = "TargetPath";
+
+        [Required]
+        public ITaskItem[] StaticWebAssets { get; set; }
+
+        [Required]
+        public ITaskItem[] WebRootFiles { get; set; }
+
+        public override bool Execute()
+        {
+            var assetsByWebRootPaths = new Dictionary<string, ITaskItem>(StringComparer.OrdinalIgnoreCase);
+            for (var i = 0; i < StaticWebAssets.Length; i++)
+            {
+                var contentRootDefinition = StaticWebAssets[i];
+                if (!EnsureRequiredMetadata(contentRootDefinition, BasePath) ||
```

Comment:
Are these guaranteed to exist on all items (including ones that we haven't created)? Returning false from the task fails the build making which makes this pertinent.

Reply:
@pranavkm We are the only ones creating static web assets items, and the contract for them is that they have the properties defined in the StaticWebAssets.targets file, otherwise they wouldn't work well.